src/fst.rs

```diff
@@ -318,80 +318,86 @@ pub(crate) fn push_zeros(vec: &mut Vec<u8>, len: usize) {
+            FstScopeType::Begin => ScopeType::Begin,
+            FstScopeType::Fork => ScopeType::Fork,
+            FstScopeType::Generate => ScopeType::Generate,
+            FstScopeType::Struct => ScopeType::Struct,
+            FstScopeType::Union => ScopeType::Union,
+            FstScopeType::Class => ScopeType::Class,
+            FstScopeType::Interface => ScopeType::Interface,
+            FstScopeType::Package => ScopeType::Package,
+            FstScopeType::Program => ScopeType::Program,
+            FstScopeType::VhdlArchitecture => ScopeType::VhdlArchitecture,
+            FstScopeType::VhdlProcedure => ScopeType::VhdlProcedure,
+            FstScopeType::VhdlFunction => ScopeType::VhdlFunction,
+            FstScopeType::VhdlRecord => ScopeType::VhdlRecord,
+            FstScopeType::VhdlProcess => ScopeType::VhdlProcess,
+            FstScopeType::VhdlBlock => ScopeType::VhdlBlock,
+            FstScopeType::VhdlForGenerate => ScopeType::VhdlForGenerate,
+            FstScopeType::VhdlIfGenerate => ScopeType::VhdlIfGenerate,
+            FstScopeType::VhdlGenerate => ScopeType::VhdlGenerate,
+            FstScopeType::VhdlPackage => ScopeType::VhdlPackage,
+            FstScopeType::AttributeBegin
+            | FstScopeType::AttributeEnd
+            | FstScopeType::VcdScope
+            | FstScopeType::VcdUpScope => unreachable!("unexpected scope type!"),
+        }
     }
 }
 
-fn convert_var_tpe(tpe: FstVarType) -> VarType {
-    match tpe {
-        FstVarType::Wire => VarType::Wire,
-        FstVarType::Event => VarType::Event,
-        FstVarType::Integer => VarType::Integer,
-        FstVarType::Parameter => VarType::Parameter,
-        FstVarType::Real => VarType::Real,
-        FstVarType::RealParameter => VarType::Parameter,
-        FstVarType::Reg => VarType::Reg,
-        FstVarType::Supply0 => VarType::Supply0,
-        FstVarType::Supply1 => VarType::Supply1,
-        FstVarType::Time => VarType::Time,
-        FstVarType::Tri => VarType::Tri,
-        FstVarType::TriAnd => VarType::TriAnd,
-        FstVarType::TriOr => VarType::TriOr,
-        FstVarType::TriReg => VarType::TriReg,
-        FstVarType::Tri0 => VarType::Tri0,
-        FstVarType::Tri1 => VarType::Tri1,
-        FstVarType::Wand => VarType::WAnd,
-        FstVarType::Wor => VarType::WOr,
-        FstVarType::Port => VarType::Port,
-        FstVarType::SparseArray => VarType::SparseArray,
-        FstVarType::RealTime => VarType::RealTime,
-        FstVarType::GenericString => VarType::String,
-        FstVarType::Bit => VarType::Bit,
-        FstVarType::Logic => VarType::Logic,
-        FstVarType::Int => VarType::Int,
-        FstVarType::ShortInt => VarType::ShortInt,
-        FstVarType::LongInt => VarType::LongInt,
-        FstVarType::Byte => VarType::Byte,
-        FstVarType::Enum => VarType::Enum,
-        FstVarType::ShortReal => VarType::ShortReal,
+impl From<FstVarType> for VarType {
```

Comment:
This also makes a private function public and adds types from the `fst-native` library to the public API of `wellen`